cpp/src/io/parquet/page_decode.cuh

```diff
@@ -149,10 +149,13 @@ inline __device__ bool is_bounds_page(page_state_s* const s,
   size_t const begin      = start_row;
   size_t const end        = start_row + num_rows;
 
-  // for non-nested schemas, rows cannot span pages, so use a more restrictive test
+  // For non-nested schemas, rows cannot span pages, so use a more restrictive test except for
+  // the page_end. This is because we may adjusted the `num_rows` for the last page in
+  // `generate_list_column_row_count_estimates()` to compensate for the list row size estimates
+  // in case of chunked read mode.
   return has_repetition
            ? ((page_begin <= begin && page_end >= begin) || (page_begin <= end && page_end >= end))
-           : ((page_begin < begin && page_end > begin) || (page_begin < end && page_end > end));
+           : ((page_begin < begin && page_end > begin) || (page_begin < end && page_end >= end));
```

Comment:
`generate_list_column_row_count_estimates()` clips the `num_rows` for the last page to compensate for list cols. This leads to a false negative `is_bounds_page` when decoding strings for the last page. This leads to a string data buffer `end_value` to include the entire last page instead of clipping it at `nrows` leading those extra `\x00` in the output buffer.

Reply:
@nvdbaranec I didn't see the effect of adjusting `num_rows` for the last page showing up for other col types and `is_bounds_page` is only used in string decoders so I am hoping other decoders don't rely on a similar predicate (at least doesn't seem like in the tests).

Reply:
You know it's a convoluted bug when the fix is 99% comment.

Should we try to keep the condition and change the`num_rows` adjustment? This fix feels like a workaround for incorrect `num_rows` (expecting to be wrong on this one, but want to make sure we at least consider this).

Reply:
Yeah, I have been debating myself with this one as well. I am not fully sure if we would break things by adjusting (to what?) `num_rows` for the last page in `generate_list_column_row_count_estimates()` and keep the original condition here. Maybe @nvdbaranec or @etseidl can comment here better. 

Currently I chose to relax this condition as this function is only used in string decoders and nowhere else and that's only where we are seeing the problem I described above.

Reply:
Thanks for the feedback, Ed! I have added a new boolean field to `PageInfo` which tells if the `num_rows` for this page has been adjusted and if so, we relax the condition.